### PR TITLE
Improve mask fit documentation

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request adapt significantly the mask_map.ipynb tutorial to explain in detail how to create a `mask_fit` and clearly distinguish from the exclusion mask.
The Fermi dataset is therefore loaded early in the notebook and we show how create a energy mask as well as a region mask and how to combine them on this dataset using directly the `dataset` `geom`.

I have introduced also two properties on the `MapDataset` for clarity to the user. One is the `.energy_range` which return the full energy range for fitting, the other is the `safe_energy_range` which returns only the safe range range. 
Note that the latter change, breaks the current definition of `energy_range`. We might possibly adapt to `fit_energy_range` so that everything is clear (although the latter might not contain the safe range...).

Comments @AtreyeeS @QRemy @adonath @bkhelifi ? Should it help users who currently simply do stuff on the `mask_safe`?

**Dear reviewer**

### CODE REVIEW

**Review by adonath — COMMENTED** [review]
Thanks @registerrier, this is really helpful! 

A few preliminary comments on the `energy_range` definition:
- Maybe we stick with the mask nomenclature like: `energy_range`, `energy_range_safe`, `energy_range_fit`
- To avoid code duplication one could introduce `Map.mask_energy_range()`

File: gammapy/datasets/map.py
Comment (AtreyeeS) [inline gammapy/datasets/map.py:367]: The doc string should read only `mask_safe`, no?

**Review by AtreyeeS — APPROVED** [review]
Thanks a lot @registerrier ! This makes the concept clear. 
I think we should expose the `mask_fit` a bit more in our tutorials, and advertise this notebook more.

Perhaps it is better to have `safe_energy_range`, `fit_energy_range` and`energy_range`(which is the intersection of the previous two) explicitly?
What is being done in the new `plot` method introduced by @fabiopintore ? We should follow a consistent nomenclature...

**Comment by adonath** [comment]
Thanks @registerrier, I think content wise the addition is very nice, however I think the order could maybe be improved. I think the tutorial should start with the basics of maps as masks, incl. creation and arithmetics, move on to more advance things like erosion and dilation and finally explain the interaction with the higher level `MapDataset` structure. From a pedagogical point of view, I think it's non ideal to start with creating `MapDataset`, when the tutorial is about masks...

On 2nd thought I actually prefer  the following: I think  we are still missing a `dataset.ipynb` API tutorial, which explains the basics of the `Dataset` classes (see https://github.com/gammapy/gammapy/issues/3143). What about adding the `.mask_fit` etc. content as the  first content to the a new API tutorial explaining datasets?

**Comment by registerrier** [comment]
> Thanks @registerrier, this is really helpful!
> 
> A few preliminary comments on the `energy_range` definition:
> 
> * Maybe we stick with the mask nomenclature like: `energy_range`, `energy_range_safe`, `energy_range_fit`
> * To avoid code duplication one could introduce `Map.mask_energy_range()`

OK. I am not sure whether it is a good idea to allow an `energy_range_fit`. This could suggest that this is the energy range that will be used for the fit. I suppose this is similar to the mask naming scheme, so it is probably fine though.

I am not found of adding methods on `Maps` that only relate to some specific types of maps (here boolean with an energy axis). Since these methods are exposed to the user, it can be quite confusing on what they can do. I would rather have it a function possibly in a file such as `gammapy/datasets/utils.py`.

**Comment by adonath** [comment]
> OK. I am not sure whether it is a good idea to allow an energy_range_fit. This could suggest that this is the energy range that will be used for the fit. I suppose this is similar to the mask naming scheme, so it is probably fine though.

A agree the choice of name is non-ideal for the man as well, but I think the consistency is more important.

> I am not found of adding methods on Maps that only relate to some specific types of maps (here boolean with an energy axis). Since these methods are exposed to the user, it can be quite confusing on what they can do. I would rather have it a function possibly in a file such as gammapy/datasets/utils.py.

As you prefer. In the helper function one could maybe even return a complete map like `energy_range_maps(mask) > (e_min_map, e_max_map)`, which returns the energy range per pixel. Could be interesting to visualise for datasets, where the threshold changes with position.

**Comment by registerrier** [comment]
After today's meeting we've decided to merge this now and adapt once the dataset API notebook is written.

**Pull request merged**
